### PR TITLE
Update returned model

### LINKED ISSUE #4: Add adapter name to returned model
Also, while working on this add the model name and tableName, so that it doesn't have to be explicitly added after models have been initialized.

### PR DESCRIPTION
**Pull request by stephen-ta**
Closes #4

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ const use = (modelName, adapterName) => {
   // Initialize model and adapter
   const m = model.init(modelName)
   const a = adapter.init(adapterName)
+  // Add adapter name to returned model
+  m.adapter = adapterName
```
Comment (stephen-ta) [inline src/index.js]: NOTE: Will be used later to check if "mysql" and then do specific tasks

File: src/lib/model.js [hunk 2]
```diff
@@ -21,8 +21,8 @@ model.store = {}
  */
 model.add = (m) => {
   // Ensure required properties
-  if (!m.name || !m.version || !m.schema) {
-    throw new Error('Model must contain a name, version and schema')
+  if (!m.name || !m.tableName || !m.version || !m.schema) {
```
Comment (stephen-ta) [inline src/lib/model.js]: NOTE: "tableName" has already been technically required by modli-dynamodb, and we've been explicitly adding it after model initialization on myqsl models, so this now makes it required and allows us to set it in our model initially without having to specify it later.

File: src/lib/model.js [hunk 3]
```diff
@@ -56,8 +56,11 @@ model.init = (m) => {
     throw new Error('Model not defined')
   }
   // Get model object
+  const defaultVersion = Object.keys(model.store[m]).pop()
   return {
-    defaultVersion: Object.keys(model.store[m]).pop(),
+    name: m,
```
Comment (stephen-ta) [inline src/lib/model.js]: Similarly, we are explicitly setting "name" after model initialization since it wasn't being set during initialization.

**Comment by Fluidbyte** [comment]
:accept:

**Pull request merged**